public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -102,6 +102,38 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
     return getBackendSrv().fetch<Record<string, any>>(req);
   }
 
+  getLogsVolumeQuery(request: DataQueryRequest<LokiQuery>): Observable<DataQueryResponse> | undefined {
+    const isLogsVolumeAvailable = request.targets.some((target) => !isMetricsQuery(target.expr) && !!target.expr);
+    if (!isLogsVolumeAvailable) {
+      return;
+    }
+
+    const histogramRequest = cloneDeep(request!);
+    histogramRequest.targets = histogramRequest.targets
+      .filter((target) => !isMetricsQuery(target.expr))
+      .map((target) => {
```

Comment:
this `map()` call mutates, and that's a little not-idiomatic in my opinion... considering you cloneDeep-ed the whole thing before, i recommend a forEach here
```.map((target) => {target.expr = `count_over_time(${target.expr}[$__interval])`;})```


or, if you want to go fancy 😁 , we do have the `immer` package in our `package.json` ( https://yarnpkg.com/package/immer ), it is very useful in such situations, you could skip the `cloneDeep` with it.

Reply:
Fixed in db6522613c3ca30193587680ed8a7a5e83ffda53